test/test_transformers.py

```diff
@@ -1505,13 +1505,98 @@ def test_sdp_math_gradcheck(self, device, contiguous_inputs: bool):
 
     @onlyCPU
     @parametrize("type", ["dense", "nested"])
-    def test_fused_sdp_choice_cpu(self, device, type: str):
+    @parametrize("dropout", [0.0, 0.7])
+    def test_fused_sdp_choice_cpu(self, device, type: str, dropout: float):
         # Test that cpu and nestedtensor cpu return MATH backend
-        for dtype in floating_types_and_half():
+        for dtype in floating_types_and_half() + (torch.bfloat16,):
             make_tensor = partial(rand_sdpa_tensor, type=type, device=device, dtype=dtype)
-            size = (2, 2, 3, 4)
+            size = (2, 128, 8, 64)
             q, k, v = make_tensor(size), make_tensor(size), make_tensor(size)
-            assert torch._fused_sdp_choice(q, k, v) == SDPBackend.MATH
+            if type == "nested" \
+                    or dropout > 0.0 \
+                    or dtype not in [torch.float32, torch.float64, torch.bfloat16]:
+                assert torch._fused_sdp_choice(q, k, v, dropout_p=dropout) == SDPBackend.MATH
+            else:
+                assert torch._fused_sdp_choice(q, k, v, dropout_p=dropout) == SDPBackend.FLASH_ATTENTION
+
+    @onlyCPU
+    @parametrize("fused_kernel", [SDPBackend.FLASH_ATTENTION])
+    @parametrize("dtype", [torch.float64, torch.float32, torch.bfloat16])
+    @parametrize("batch_size", [2, 12])
+    @parametrize("seq_len", [267, 1030])
+    @parametrize("n_head", [1, 3])
+    @parametrize("head_dim", [8, 16])
+    @parametrize("causal", [True, False])
+    @parametrize("train", [True, False])
+    def test_scaled_dot_product_fused_attention_vs_math_cpu(
+        self,
+        device,
+        fused_kernel,
+        dtype,
+        batch_size,
+        seq_len,
+        n_head,
+        head_dim,
+        causal,
+        train,
+    ):
+        atol = 1e-5
+        rtol = 5e-6
+        if dtype is torch.bfloat16:
+            atol = 1e-2
+            rtol = 1e-2
+        if dtype is torch.bfloat16 and causal and train:
+            atol = 2e-2
+            rtol = 2e-2
```

Comment:
Is it possible to use some of the same helpers for creating inputs defined here: 
https://github.com/pytorch/pytorch/blob/b08b0c915f59cce79630a027fe501f54b6b15e4f/test/test_transformers.py#L101